### PR TITLE
Test parallelize python tests

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

#2003 Test whether parallelize will have problems.
#2018 Parallelize may fasten the speed.

### CODE REVIEW

File: pyspark/test/dev/run-tests [hunk 1]
```diff
@@ -28,13 +28,13 @@ do
...
     export PYSPARK_DRIVER_PYTHON=$p
-    $p -m pytest -v --doctest-modules ../../../pyspark/bigdl \
+    $p -m pytest -v -n 4 --doctest-modules ../../../pyspark/bigdl \
     --ignore=../../../pyspark/bigdl/dataset/ \
     --ignore=../../../pyspark/bigdl/util/tf_utils.py \
     --ignore=../../../pyspark/bigdl/keras \
     --ignore=../../../pyspark/test/bigdl/keras/test_application.py \
     --ignore=../../../pyspark/bigdl/models/ && \
-    $p -m pytest -v ../../../pyspark/test/ \
+    $p -m pytest -v -n 4 ../../../pyspark/test/ \
```
Comment (zhichao-li) [inline pyspark/test/dev/run-tests]: modify run-keras as well

**Comment by zhichao-li** [comment]
add `keras.backend.set_image_dim_ordering("th")` to `test_lenet`

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V-P/job/BigDL-PR-V/482/

**Comment by hkvision** [comment]
It still causes random failure http://172.168.2.101:8080/job/BigDL-PR-V/491/
cc @zhichao-li

**Comment by hkvision** [comment]
After using `-n 4` for keras test, time will be reduced from ~32min to ~11min.

**Comment by hkvision** [comment]
failed cases: http://172.168.2.101:8080/job/BigDL-PR-V/524/ http://172.168.2.101:8080/job/BigDL-PR-V/517/

**Pull request closed without merging**